Integrations/integration-MCAS.yml

```diff
@@ -0,0 +1,265 @@
+commonfields:
+  id: Microsoft Cloud App Security
+  version: -1
+name: Microsoft Cloud App Security
+display: Microsoft Cloud App Security
+category: Network Security
+image: data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAbcAAAG3BAMAAADFuQbYAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAAElBMVEXz8/PzUyWBvAYFpvD/ugj///98Xc9SAAAAAWJLR0QF+G/pxwAAAAd0SU1FB+IECxAWCcy7n+sAAAGSSURBVHja7c9BCQAgEABBxQJawQSCFeyfyZcRTu4xW2CZUiRJkiRJylodwfV3mtHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcHBwcF9wUmSJEmSlK62g1vvdKKDg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4ODg4P7gpMkSZIkKUsX5n/TJVBjPXQAAAAldEVYdGRhdGU6Y3JlYXRlADIwMTgtMDQtMTFUMTY6MjI6MDkrMDA6MDD3dzx7AAAAJXRFWHRkYXRlOm1vZGlmeQAyMDE4LTA0LTExVDE2OjIyOjA5KzAwOjAwhiqExwAAAABJRU5ErkJggg==
+description: Microsoft Cloud App Security (MCAS)
+configuration:
+- display: Server URL
+  name: url
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: API Token
+  name: token
+  defaultvalue: ""
+  type: 4
+  required: true
+- display: Ignore server certificate
+  name: insecure
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Use System Proxy
+  name: proxy
+  defaultvalue: ""
+  type: 8
+  required: false
+script:
+  script: |-
+    import requests
+    import json
+    import yaml
+
+    SERVER = demisto.params()['url'][:-1] if demisto.params()['url'].endswith('/') else demisto.params()['url']
+    TOKEN = demisto.params().get('token')
+    USE_SSL = not demisto.params().get('insecure', False)
+
+    if not demisto.params()['proxy']:
+        del os.environ['HTTP_PROXY']
+        del os.environ['HTTPS_PROXY']
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+    def mcas_list_alerts():
+        limit = demisto.args()['limit']
+        severity_raw = demisto.args()['severity']
+        if severity_raw == "Low":
+            severity = "0"
+        elif severity_raw == "Medium":
+            severity = "1"
+        elif severity_raw == "High":
+            severity = "2"
+        else:
+            severity = "2,1,0"
+        url = SERVER + '/cas/api/v1/alerts/'
+        data = '{"skip":0,"limit":' + limit + ',"filters":{"resolutionStatus":{"eq":[0]},"severity":{"eq":[' + severity + ']}},"sortField":"date","sortDirection":"desc","performAsyncTotal":true}'
+        headers = {'Authorization': 'Token ' + TOKEN, 'Content-type': 'application/json'}
+        r = requests.post(url=url, headers=headers, data=data, verify=USE_SSL)
+        j = yaml.load(json.dumps(r.json()))
```

Comment:
It is a bit risky to call `.json()` like that, as it might raise an encoding exception.
It is better to wrap it in `try-except`, catching `ValueError`

Reply:
Updated in offline copy; will upload after subsequent changes fixed.